### PR TITLE
Wiring up new homepage to new dashboard

### PR DESCRIPTION
**Pull request by benwerd**

### CODE REVIEW

File: unlock-app/src/components/creator/Dashboard.js [hunk 1]
```diff
@@ -4,9 +4,15 @@ import React, { Component } from 'react'
...
 import CreatorLocks from './CreatorLocks'
+import {setTransaction} from '../../actions/transaction'
+import { connect } from 'react-redux'
 
-export default class Dashboard extends Component {
+export class Dashboard extends Component {
   render() {
+    if (!this.props.account) {
+      return null //loading
+    }
```
Comment (julien51) [inline unlock-app/src/components/creator/Dashboard.js]: We should probably eventually handle that at a higher level.. But that'll do for now!

File: unlock-app/src/components/creator/Dashboard.js [hunk 2]
```diff
@@ -22,3 +28,18 @@ Dashboard.propTypes = {
...
+  return {
+    account: state.network.account, // TODO change account to base level
+    network: state.network,
+    transactions: state.transactions,
+    locks: state.locks,
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  setTransaction: (transaction) => dispatch(setTransaction(transaction)),
```
Comment (julien51) [inline unlock-app/src/components/creator/Dashboard.js]: Are we actually calling this anywhere in the component?

File: unlock-app/src/components/pages/Home.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
 
 export default class Home extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.dashboardClick = this.dashboardClick.bind(this)
+  }
+
+  dashboardClick() {
+    this.props.history.push('/dashboard')
+  }
```
Comment (julien51) [inline unlock-app/src/components/pages/Home.js]: Hum, I would expect react router to handle that for you?

File: unlock-app/src/components/pages/Home.js [hunk 4]
```diff
@@ -13,7 +23,7 @@ export default class Home extends PureComponent {
           Unlock is a protocol which enables creators to monetize their content with a few lines of code in a fully decentralized way.
         </Headline>
 
-        <Action>
+        <Action onClick={this.dashboardClick}>
           <HomepageButton>Go to Your Dashboard</HomepageButton>
```
Comment (julien51) [inline unlock-app/src/components/pages/Home.js]: Same here, you should probably use `Link` from react-router-dom.

**Comment by benwerd** [comment]
Yep, I'm working on the changes!

**Comment by benwerd** [comment]
Using router. PTAL

File: unlock-app/src/components/pages/Home.js [hunk 5]
```diff
@@ -13,10 +15,12 @@ export default class Home extends PureComponent {
           Unlock is a protocol which enables creators to monetize their content with a few lines of code in a fully decentralized way.
         </Headline>
 
-        <Action>
-          <HomepageButton>Go to Your Dashboard</HomepageButton>
-          <ButtonLabel>Requires a browser with an Ethereum wallet</ButtonLabel>
-        </Action>
+        <Link to={'/dashboard'}>
```
Comment (julien51) [inline unlock-app/src/components/pages/Home.js]: You'll know what I say about nesting components!... but I guess you tried without?

File: unlock-app/src/components/pages/Home.js [hunk 6]
```diff
@@ -51,6 +55,10 @@ export default class Home extends PureComponent {
   }
 }
 
+Home.propTypes = {
```
Comment (julien51) [inline unlock-app/src/components/pages/Home.js]: You can probably get rid of this ;)

File: unlock-app/src/components/pages/Home.js (hunk 5 above)
Comment (benwerd) [reply unlock-app/src/components/pages/Home.js]: I did, yeah. I have some feature requests for the next version of styled-components.

Comment (julien51) [reply unlock-app/src/components/pages/Home.js]: lol. happy to hear them!

**Pull request merged**